nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/html/HtmlPythonProcessorDocumentationWriterTest.java

```diff
@@ -0,0 +1,187 @@
+import static org.apache.nifi.documentation.html.XmlValidator.assertContains;
+import static org.apache.nifi.documentation.html.XmlValidator.assertNotContains;
+import static org.mockito.Mockito.when;
+
+public class HtmlPythonProcessorDocumentationWriterTest {
+
+    @Test
+    public void testProcessorDocumentation() throws IOException {
+        final PythonProcessorDetails processorDetails = getPythonProcessorDetails();
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        final DocumentationWriter<PythonProcessorDetails> writer = new HtmlPythonProcessorDocumentationWriter();
+        writer.write(processorDetails, outputStream, false);
+
+        final String results = outputStream.toString();
+        XmlValidator.assertXmlValid(results);
+
+        assertContains(results, "This is a test capability description");
+        assertContains(results, "tag1, tag2, tag3");
+
+        final List<PropertyDescription> propertyDescriptions = getPropertyDescriptions();
+        propertyDescriptions.forEach(propertyDescription -> {
+            assertContains(results, propertyDescription.getDisplayName());
+            assertContains(results, propertyDescription.getDescription());
+            assertContains(results, propertyDescription.getDefaultValue());
+        });
+
+        assertContains(results, "Supports Expression Language: true (will be evaluated using Environment variables only)");
+        assertContains(results, "Supports Expression Language: true (will be evaluated using flow file attributes and Environment variables)");
+
+        final List<UseCaseDetails> useCases = getUseCases();
+        useCases.forEach(useCase -> {
+            assertContains(results, useCase.getDescription());
+            assertContains(results, useCase.getNotes());
+            assertContains(results, String.join(", ", useCase.getKeywords()));
+            assertContains(results, useCase.getConfiguration());
+        });
+
+        final List<MultiProcessorUseCaseDetails> multiProcessorUseCases = getMultiProcessorUseCases();
+        multiProcessorUseCases.forEach(multiProcessorUseCase -> {
+            assertContains(results, multiProcessorUseCase.getDescription());
+            assertContains(results, multiProcessorUseCase.getNotes());
+            assertContains(results, String.join(", ", multiProcessorUseCase.getKeywords()));
+
+            multiProcessorUseCase.getConfigurations().forEach(configuration -> {
+                assertContains(results, configuration.getProcessorType());
+                assertContains(results, configuration.getConfiguration());
+            });
+        });
+
+        assertNotContains(results, NO_PROPERTIES);
+        assertNotContains(results, "No description provided.");
+        assertNotContains(results, NO_TAGS);
+    }
+
+    @Test
+    public void testEmptyProcessor() throws IOException {
+        final PythonProcessorDetails processorDetails = Mockito.mock(PythonProcessorDetails.class);
```

Comment:
The `Mockito.mock()` calls can be replaced with a static import so that the shorter `mock()` call can be used.